packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -442,10 +453,19 @@ class AttachDisk extends Component {
         const vbd = vbds[id]
         return !vbd || !vbd.attached || vbd.read_only
       })
-    return attachDiskToVm(vdi, vm, {
-      bootable,
-      mode: readOnly || !_isFreeForWriting(vdi) ? 'RO' : 'RW',
-    }).then(onClose)
+
+    const _attachDisktoVm = () =>
+      attachDiskToVm(vdi, vm, {
+        bootable,
+        mode: readOnly || !_isFreeForWriting(vdi) ? 'RO' : 'RW',
+      }).then(onClose)
+
+    return this._checkSr()
+      ? _attachDisktoVm()
+      : confirm({
+          title: _('attachDisk'),
+          body: _('warningVdiSr'),
+        }).then(_attachDisktoVm)
```

Comment:
Let's keep it homogeneous with the "New disk" feature and show the warning right below the select instead of in a modal.